# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/libkscreen/hotfix-wayland.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- libkscreen-6.0.4/backends/kwayland/waylandoutputdevice.h.vanilla	2024-04-28 14:55:30.945000000 +0200
+++ libkscreen-6.0.4/backends/kwayland/waylandoutputdevice.h	2024-04-28 14:56:49.643000000 +0200
@@ -91,9 +91,9 @@
     void kde_output_device_v2_wide_color_gamut(uint32_t wcg_enabled) override;
     void kde_output_device_v2_auto_rotate_policy(uint32_t policy) override;
     void kde_output_device_v2_icc_profile_path(const QString &profile) override;
-    void kde_output_device_v2_brightness_metadata(uint32_t max_peak_brightness, uint32_t max_frame_average_brightness, uint32_t min_brightness) override;
-    void kde_output_device_v2_brightness_overrides(int32_t max_peak_brightness, int32_t max_average_brightness, int32_t min_brightness) override;
-    void kde_output_device_v2_sdr_gamut_wideness(uint32_t value) override;
+    void kde_output_device_v2_brightness_metadata(uint32_t max_peak_brightness, uint32_t max_frame_average_brightness, uint32_t min_brightness);
+    void kde_output_device_v2_brightness_overrides(int32_t max_peak_brightness, int32_t max_average_brightness, int32_t min_brightness);
+    void kde_output_device_v2_sdr_gamut_wideness(uint32_t value);
 
 private:
     QString modeName(const WaylandOutputDeviceMode *m) const;
--- libkscreen-6.0.4/backends/kwayland/waylandoutputdevice.cpp.vanilla	2024-04-28 14:57:58.875000000 +0200
+++ libkscreen-6.0.4/backends/kwayland/waylandoutputdevice.cpp	2024-04-28 14:59:57.321000000 +0200
@@ -304,6 +304,7 @@
         wlConfig->set_icc_profile_path(object(), output->iccProfilePath());
         changed = true;
     }
+#if 0
     if (kde_output_configuration_v2_get_version(wlConfig->object()) >= KDE_OUTPUT_CONFIGURATION_V2_SET_SDR_GAMUT_WIDENESS_SINCE_VERSION
         && m_sdrGamutWideness != output->sdrGamutWideness()) {
         wlConfig->set_sdr_gamut_wideness(object(), std::clamp<uint32_t>(std::round(output->sdrGamutWideness() * 10'000), 0, 10'000));
@@ -318,6 +319,7 @@
                                            std::round(output->minBrightnessOverride().value_or(-0.000'1) * 10'000.0));
         changed = true;
     }
+#endif
 
     return changed;
 }
